src/core/ngx_palloc.c

```diff
@@ -183,6 +173,8 @@ ngx_palloc_small(ngx_pool_t *pool, size_t size, ngx_uint_t align)
             p->d.last = m + size;
 
 #ifdef __CHERI_PURE_CAPABILITY__
+            // This is safe so long as NGX_MAX_ALLOC_FROM_POOL is
+            // representable.  By default it's 4095 so safe.
```

Comment:
Maybe add a _Stativ_assert()? Unfortunately I don't think the builtins can be used in constant expressions yet. But that should be easy to add if we think it's useful.

Reply:
Unfortunately NGX_MAX_ALLOC_FROM_POOL isn't static (it's `(ngx_pagesize - 1)`). I could add an assert in `ngx_create_pool`, but nginx doesn't believe in asserts so think I'll leave this alone for now.